update_gofish.sh

```diff
@@ -37,6 +45,6 @@ else
 fi
 
 # Merge back into main
-git checkout main
+git checkout "$PR_BRANCH_NAME"
```

Comment:
which remote is this checking out? this should be getting a fresh copy of upstream/main and then applying our patch on top of that, then pushing to internal/PR_BRANCH_NAME.

Reply:
is it wrong as-is now?

so:
- line 30: checkout main (should be origin AKA internal main)
- 31: create the PR branch locally
- fetch upstream and rebase it onto the PR branch
- still on PR branch
- 37: create a new apply branch to apply our patch
- 48: come back to our local PR branch (already rebased onto upstream)
- merge the patch back in again

the logic in the script was a bit confusing to me, but AFAIK, the only thing I changed was instead of rebasing `origin/main` we rebase `$PR_BRANCH` and then push that up to `origin/$PR_BRANCH`

Reply:
i think thats right. perhaps if we renamed the remotes to `upstream` and `internal`, it might make its more clear/explicit